packages/desktop/src/renderer/components/widgets/channels/UploadedFilesPreviews.tsx

```diff
@@ -0,0 +1,193 @@
+import { useModal } from '../../../containers/hooks'
+import { ModalName } from '../../../sagas/modals/modals.types'
+
+export interface FilePreviewData {
+  [id: string]: FileContent
+}
+
+export interface FilePreviewComponentProps {
+  fileData: FileContent
+  onClick: () => void
+}
+
+const useStyles = makeStyles(() => ({
+  inputFiles: {
+    display: 'flex',
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    justifyContent: 'flexStart',
+    alignItems: 'baseline',
+    alignContent: 'stretch',
+    paddingRight: '50px'
+  },
+  image: {
+    width: '64px',
+    height: '64px',
+    borderRadius: '15%',
+    marginLeft: '10px',
+    objectFit: 'cover'
+  },
+  closeIconContainer: {
+    position: 'absolute',
+    margin: '0',
+    padding: '0',
+    right: '0px',
+    top: '0px',
+    backgroundColor: 'white',
+    borderRadius: '100%',
+    width: '22px',
+    height: '22px',
+    transform: 'translate(50%, -50%)',
+    '&:hover': {
+      backgroundColor: '#dddddd'
+    }
+  },
+  closeIcon: {
+    position: 'relative',
+    left: '50%',
+    top: '50%',
+    color: '#444444',
+    transform: 'translate(-50%, -50%)',
+    '&:hover': {
+      color: '#000000'
+    },
+    width: '17px'
+  },
+  imageContainer: {
+    position: 'relative',
+    cursor: 'pointer'
+  },
+  tooltip: {
+    marginTop: '8px'
+  }
+}))
+
+const FilePreviewComponent: React.FC<FilePreviewComponentProps> = ({ fileData, onClick }) => {
+  const [showClose, setShowClose] = useState(false)
+  const classes = useStyles({})
+  return (
+    <div
+      className={classes.imageContainer}
+      onMouseLeave={() => {
+        setShowClose(false)
+      }}
+      onMouseOver={() => {
+        setShowClose(true)
+      }}>
+      {showClose && (
+        <Tooltip title='Remove' placement='top' className={classes.tooltip}>
+          <div className={classes.closeIconContainer} onClick={onClick}>
+            <CloseIcon className={classes.closeIcon} />
+          </div>
+        </Tooltip>
+      )}
+
+      <img src={fileData.path} alt={fileData.name} className={classes.image} />
+    </div>
+  )
+}
+
+export interface UploadFilesPreviewsProps {
+  filesData: FilePreviewData
+  removeFile: (id: string) => void
+}
+
+enum unsuportedFileTitle {
+  singleFile = 'File unsupported',
+  someFiles = 'Some files unsuported'
+}
+
+enum unsuportedFileContent {
+  singleFileUnsupported = ' is not a file type Quiet supports.',
+  someFilesUnsupported = 'some of these file types are not supported by Quiet:',
+
+  tryUploadZip = 'Try uploading a .zip of this instead.',
+
+  sendOtherFile = 'The other file will be sent',
+  sendOtherFiles = 'The other files will be sent'
+}
+
+const supportedFilesExtensions = ['.jpg', '.jpeg', '.png']
+
+const checkAreFilesSupported = (filesData: FilePreviewData) => {
+  const unsupportedFiles: FileContent[] = []
+
+  Object.entries(filesData).map(fileData => {
+    if (!supportedFilesExtensions.includes(fileData[1].ext)) {
```

Comment:
Why index here is fixed? Does fileData always have just one item? Then why is it an array/dictionary

Reply:
It's not one item, fileData is just a `[key, value]` because we're iterating over Object.entries

Reply:
it is what ` Object.entries` do